src/api.ts

```diff
@@ -51,6 +58,7 @@ export async function applyTextEditorOptions(
 		return
 	}
 
+	commands.executeCommand('editor.action.detectIndentation') // <--- HACK!!
```

Comment:
So this is the problem I am having. If we set the `tabSize` option, it overrides both tab width and indent size, and indentation detection for some reason won't kick in.

But if we manually run indentation detection before setting the option, it works as expected, i.e. setting `tabSize` won't change `indentSize`.

(This is a hack to illustrate the problem, not an intended solution)

Reply:
Ok I found a solution, and an upstream bug.

Reply:
Bug is fixed upstream!